### DISCUSSION TITLE
search string with spaces doesn't yield

### DISCUSSION THREAD

**Discussion by anjanesh** [krisk/Fuse#769] [Q&A] [answered]
```javascript
search(value) {
if (value.length > 1) {
    const options = {
        shouldSort: true,
        keys: keys,
        useExtendedSearch: true,
        threshold: 0,
        findAllMatches: true
    }
    const fuse = new Fuse(listData, options);
    this.items = fuse.search("'" + value).map(elem => elem.item);
} else {
    this.items = listData;
}
```

For a string **ABCD FINANCE LIMITED** it searches perfectly when **ABCD** is typed but for **ABCD F** or **ABCD FINANCE LIMITED** it doesnt show in the result.

**Comment by krisk** [comment, marked as answer]
Late reply, sorry.

The include-match operator (`'`) does exact substring matching, so `'ABCD F` looks for the literal substring "ABCD F" in the field values. Combined with `threshold: 0` (perfect match only), this is very strict.

A few things to check:

1. Make sure the data actually contains `"ABCD FINANCE LIMITED"` as-is (no extra whitespace, non-breaking spaces, etc.).
2. If you want each word matched independently, you can split the input and join with spaces — in extended search, space-separated terms are treated as AND: `'ABCD 'FINANCE` will match items containing both substrings.

Something like:

```js
const query = value.split(/\s+/).map(term => "'" + term).join(" ")
this.items = fuse.search(query).map(elem => elem.item)
```

**Comment by krisk** [comment]
The `feat/token-search` branch addresses this. With `useTokenSearch: true`, "ABCD FINANCE LIMITED" is split into three independent fuzzy searches, so partial queries like "ABCD F" will match correctly.
